kv.go

```diff
@@ -91,21 +86,76 @@ type KeyValueStatus interface {
 // KeyWatcher is what is returned when doing a watch.
 type KeyWatcher interface {
 	// Updates returns a channel to read any updates to entries.
+	//
+	// A nil value is sent over the channel indicating that the end of initial data was reached,
+	// initial data may be the last available message or an indication that no messages are available
+	// at the time the watch was created.
 	Updates() <-chan KeyValueEntry
 	// Stop will stop this watcher.
 	Stop() error
 }
 
-type WatchOpt interface {
-	configureWatcher(opts *watchOpts) error
+// KeyValueOption configures the Key-Value store instance
+type KeyValueOption func(*kvs) error
+
+// BucketSubjectPrefix sets a custom prefix to use for write operations like Put() to the bucket.
+// Usually keys are accessed as $KV.BUCKET.KEY, this would replace the $KV.BUCKET. part of
+// that subject.
```

Comment:
Just read this again. This comment says replace `$KV.<BUCKET>.` but @matthiashanel told me something different, that is to put the js prefix in front of the key, so for publishes, we publish to `<prefix>.$KV.<BUCKET>.<KEY>` and the server strips the prefix. So maybe we need to wait until he's back to settle on this.

Reply:
No, this is what we are dong. We are supporting ARBITRARY prefixes, not the conventions in ADR-19. Arbitrary means any random prefix users might choose.

See comments from Mathias here https://github.com/nats-io/nats-architecture-and-design/pull/82

ADR-19 should not have been merged tbh, its way too early and the design there is not going to work. So this is what we are doing so KV can work cross account.

Reply:
Matthias and I had a slack conversation on the afternoon of the 23rd and conversation with Alberto on the 24th. I have unit test that works cross account against server v2.6.6 based on those conversations. Alberto has also coded the same as me, he is the one who provided the inbox prefix config. So, I removed my review request. I'm just going leave the java pr un-merged for now and work on something else.

Reply:
Before those called Mathias and I agreed on arbitrary prefixes, if he agreed with you on non arbitrary prefixes I am afraid that's not something I can talk about. I know what was agreed as to finalise KV 1.0 before we all went on holiday and it was not `$KV.`

Reply:
in the java code I believe `toKeySubjectConsiderPrefix` implements only the jetstream context prefix option, not an arbitrary prefix - we discussed that, you have no support for arbitrary prefixes.